src/bot.ts

```diff
@@ -0,0 +1,323 @@
+import assert from 'assert'
+import * as util from './util'
+import { Wallet } from './hathor'
+
+class Odds {
+  readonly probability: number;
+  readonly multiplier: number;
+  readonly minBet: number;
+  readonly maxBet: number;
+
+  constructor(mul: number, min: number, max: number) {
+    this.probability = 1 / mul
+    this.multiplier = mul
+    this.minBet = min
+    this.maxBet = max
+    // TODO: support other types of dices
+    assert(this.probability === util.trunc(this.probability, 2), 'multiplier should be valid for dice100')
+  }
+
+  get percentProb(): string {
+    return (this.probability * 100).toFixed(2)
+  }
+
+  get minRoll(): number {
+    return Math.floor(this.probability * 100)
+  }
+
+  prize(bet: number): number {
+    return bet * this.multiplier
+  }
+}
+
+const ODDS = [
+  // TODO: make a list that makes sense
+  new Odds(2, 1, 100),
+  new Odds(10, 1, 10),
+  new Odds(100, 1, 2)
+]
+
+function getOdds(mul: string): Odds | undefined {
+  for (const odds of ODDS) {
+    if (`${odds.multiplier}x` === mul) {
+      return odds
+    }
+  }
+}
+
+export abstract class Conversation {
+  readonly bot: Bot;
+
+  constructor(bot: Bot) {
+    this.bot = bot
+  }
+
+  abstract get content(): string;
+  abstract isDirect(): boolean;
+  abstract isPublicDedicated(): boolean;
+  abstract isFromSelf(): boolean;
+  abstract getUid(): string;
+  abstract async respond(response: string): Promise<void>;
+  abstract formatCmd(cmd: string): string;
+  shouldRespond(): boolean { return (this.isDirect() || this.isPublicDedicated()) && !this.isFromSelf() }
+
+  async handleCmd(): Promise<void> {
+    if (!this.shouldRespond()) {
+      return
+    }
+    const P = this.bot.prefixChar
```

Comment:
Maybe rename it to `prefixChar`? It's better to understand in the code

Reply:
Instead of `P`? I only used it as a shortcut, if I were to use `prefixChar` instead I could as well use `this.bot.prefixChar` directly.

Reply:
Yeah, because it was not clear to me what was `P` when I was reading the code. I had to go back to the definition more than once. But it's also ok if you want to keep that way.